src/backend/machobj.c

```diff
@@ -2310,6 +2310,7 @@ void MachObj::addrel(int seg, targ_size_t offset, symbol *targsym,
         unsigned targseg, int rtype, int val)
 {
     Relocation rel;
+    memset(&rel, 0, sizeof(rel));
```

Comment:
`Relocation` has padding bytes, which would otherwise be filled with garbage and written to disk. Garbage in output make it difficult to compare created objects and prohibit bit-identical builds. This is definitely not redundant and Valgrind is right to complain about it.